Microsoft.Toolkit.Uwp/Helpers/ConnectionHelper.cs

```diff
@@ -106,5 +109,54 @@ public static ConnectionType ConnectionType
                 return connectionType;
             }
         }
+
+    /// <summary>
+    /// Gets connection ssid for the current Wifi Connection.
+    /// </summary>
+    /// <returns> string value of current ssid/></returns>
+    ///
+    public static async Task<string> GetNetworkSsidAsync()
+        {
+            var pProfileFilter = new ConnectionProfileFilter();
+            System.Collections.Generic.IReadOnlyList<ConnectionProfile> x = await NetworkInformation.FindConnectionProfilesAsync(pProfileFilter);
+
+            try
+            {
+                WiFiAdapter firstAdapter;
+                var access = await WiFiAdapter.RequestAccessAsync();
+                if (access != WiFiAccessStatus.Allowed)
+                {
+                   throw new UnauthorizedAccessException("Acess Denied for wifi Control");
+                }
+                else
+                {
+                    var result = await Windows.Devices.Enumeration.DeviceInformation.FindAllAsync(WiFiAdapter.GetDeviceSelector());
+                    if (result.Count >= 1)
+                    {
+                        firstAdapter = await WiFiAdapter.FromIdAsync(result[0].Id);
+                    }
+                    else
+                    {
+                        throw new NullReferenceException("No WiFi Adapters detected on this machine");
+                    }
+
+                    var connectedProfile = await firstAdapter.NetworkAdapter.GetConnectedProfileAsync();
+                    if (connectedProfile != null)
+                    {
+                        return connectedProfile.ProfileName;
+                    }
+                    else if (connectedProfile == null)
+                    {
+                        throw new NullReferenceException("WiFi adapter disconnected");
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                throw e;
```

Comment:
Nothing, presumably. No point catching and throwing it again, just let it happen and be handled further up.